packages/core/src/components/i-controller.ts

```diff
@@ -0,0 +1,3 @@
+export interface IController<T> {
```

Comment:
I'm for having interfaces in a separated file for discover-ability, but a file per interface could be too wild. Let's group all interfaces in a file name either `definitions` (v2) or `interfaces` (v1)